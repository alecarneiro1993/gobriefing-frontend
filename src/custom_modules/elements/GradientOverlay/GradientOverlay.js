import styled from 'styled-components';
import Box from '@material-ui/core/Box';

import { CENTER, FULL, FLEX } from 'utils/constants';

const GradientOverlay = styled(Box)`
  background: ${({ color: { from, to } }) => `linear-gradient(${from}, ${to});`}
  height: ${({ height }) => (height ? `${height}vh` : `${FULL}vh`)} ;
  display: ${FLEX};
  align-items: ${CENTER};
  justify-content: ${CENTER};
`;

export default GradientOverlay;
