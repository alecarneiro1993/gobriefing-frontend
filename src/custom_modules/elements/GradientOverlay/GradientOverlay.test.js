import React from 'react';
import { shallow } from 'enzyme';

import GradientOverlay from '.';

describe('GradientOverlay', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<GradientOverlay />);
    expect(wrapper).toMatchSnapshot();
  });
});
