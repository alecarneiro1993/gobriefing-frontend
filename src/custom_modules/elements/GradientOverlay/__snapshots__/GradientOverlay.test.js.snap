// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientOverlay renders without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          "
  height: ",
          "100",
          "vh;
  display: ",
          "flex",
          ";
  align-items: ",
          "center",
          ";
  justify-content: ",
          "center",
          ";
",
        ],
      },
      "displayName": "Styled(Styled(MuiBox))",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "alignContent": [Function],
        "alignItems": [Function],
        "alignSelf": [Function],
        "bgcolor": [Function],
        "border": [Function],
        "borderBottom": [Function],
        "borderColor": [Function],
        "borderLeft": [Function],
        "borderRadius": [Function],
        "borderRight": [Function],
        "borderTop": [Function],
        "bottom": [Function],
        "boxShadow": [Function],
        "children": [Function],
        "className": [Function],
        "clone": [Function],
        "color": [Function],
        "component": [Function],
        "css": [Function],
        "display": [Function],
        "displayPrint": [Function],
        "flex": [Function],
        "flexBasis": [Function],
        "flexDirection": [Function],
        "flexGrow": [Function],
        "flexShrink": [Function],
        "flexWrap": [Function],
        "fontFamily": [Function],
        "fontSize": [Function],
        "fontStyle": [Function],
        "fontWeight": [Function],
        "height": [Function],
        "justifyContent": [Function],
        "justifyItems": [Function],
        "justifySelf": [Function],
        "left": [Function],
        "letterSpacing": [Function],
        "lineHeight": [Function],
        "m": [Function],
        "margin": [Function],
        "marginBottom": [Function],
        "marginLeft": [Function],
        "marginRight": [Function],
        "marginTop": [Function],
        "marginX": [Function],
        "marginY": [Function],
        "maxHeight": [Function],
        "maxWidth": [Function],
        "mb": [Function],
        "minHeight": [Function],
        "minWidth": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "order": [Function],
        "overflow": [Function],
        "p": [Function],
        "padding": [Function],
        "paddingBottom": [Function],
        "paddingLeft": [Function],
        "paddingRight": [Function],
        "paddingTop": [Function],
        "paddingX": [Function],
        "paddingY": [Function],
        "pb": [Function],
        "pl": [Function],
        "position": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "right": [Function],
        "textAlign": [Function],
        "textOverflow": [Function],
        "top": [Function],
        "visibility": [Function],
        "whiteSpace": [Function],
        "width": [Function],
        "zIndex": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Styled(MuiBox)",
        "propTypes": Object {
          "alignContent": [Function],
          "alignItems": [Function],
          "alignSelf": [Function],
          "bgcolor": [Function],
          "border": [Function],
          "borderBottom": [Function],
          "borderColor": [Function],
          "borderLeft": [Function],
          "borderRadius": [Function],
          "borderRight": [Function],
          "borderTop": [Function],
          "bottom": [Function],
          "boxShadow": [Function],
          "children": [Function],
          "className": [Function],
          "clone": [Function],
          "color": [Function],
          "component": [Function],
          "css": [Function],
          "display": [Function],
          "displayPrint": [Function],
          "flex": [Function],
          "flexBasis": [Function],
          "flexDirection": [Function],
          "flexGrow": [Function],
          "flexShrink": [Function],
          "flexWrap": [Function],
          "fontFamily": [Function],
          "fontSize": [Function],
          "fontStyle": [Function],
          "fontWeight": [Function],
          "height": [Function],
          "justifyContent": [Function],
          "justifyItems": [Function],
          "justifySelf": [Function],
          "left": [Function],
          "letterSpacing": [Function],
          "lineHeight": [Function],
          "m": [Function],
          "margin": [Function],
          "marginBottom": [Function],
          "marginLeft": [Function],
          "marginRight": [Function],
          "marginTop": [Function],
          "marginX": [Function],
          "marginY": [Function],
          "maxHeight": [Function],
          "maxWidth": [Function],
          "mb": [Function],
          "minHeight": [Function],
          "minWidth": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "order": [Function],
          "overflow": [Function],
          "p": [Function],
          "padding": [Function],
          "paddingBottom": [Function],
          "paddingLeft": [Function],
          "paddingRight": [Function],
          "paddingTop": [Function],
          "paddingX": [Function],
          "paddingY": [Function],
          "pb": [Function],
          "pl": [Function],
          "position": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
          "right": [Function],
          "textAlign": [Function],
          "textOverflow": [Function],
          "top": [Function],
          "visibility": [Function],
          "whiteSpace": [Function],
          "width": [Function],
          "zIndex": [Function],
        },
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
